pokemon/src/main/java/gamarket/MoveCollection.java

```diff
@@ -0,0 +1,135 @@
+package gamarket;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * Represents all the moves in the Pokemon universe. 
+ * A Pokemon will be able to hold at most 4 moves in its move-set. 
+ */
+public class MoveCollection {
+	
+	/** The current number of MOVEs in the array */
+	private int numMoves; 
+	
+	/** The array to contain all the MOVEs in the Pokemon Universe*/
+	private Move moveArray[]; 
+	
+	/**
+	 * The MoveCollection constructor.
+	 * It sets the numMoves to zero. 
+	 * It allocates a new array of Move type to 7. 
+	 * 
+	 */
+	public MoveCollection()
+	{
+		numMoves = 0; 
+		moveArray = new Move[7];
+	}
+	
+	/**
+	 * The addOrModifyMove fucntion of type void.
+	 * The function adds a new move to the move collection, if the new move does not exist.
+	 * If it does exist, the move information will get updated. 
+	 * @param name This is the Move's name. Should have the Move's name(Tackle). 
+	 * @param type This is the Move's type. Should have the Move's type(Normal). 
+	 * @param category This is the Move's category. Should have the Move's category(Physical).
+	 * @param damage This is the Move's damage. Should have the Move's damage(60). 
+	 */
+	public void addOrModifyMove(String name, String type, String category, String damage)
+	{
+		try {	// running time error prevention.   
+				// double the size of the array if the current array is full. 
+			if(numMoves == moveArray.length-1)
+			{
+				moveArray = Arrays.copyOf(moveArray, moveArray.length * 2);
+			}
+				
+			// change the Strings back into type Integer. 
+			int myDmg = Integer.parseInt(damage);
+			boolean moveExists = false; 
+				
+			// traverse the array to see if the MOVE already "exists" in the collection.
+			for(int i = 0; i < numMoves && moveExists != true; i++)
+				{
+					// if both MOVEs are equal, then the MOVE "exists". 
+					if(moveArray[i].getMoveName().compareToIgnoreCase(name) == 0)
+					{
+						// proceed to modify the move-type, move-type-category, move-damage. 
+						moveArray[i].setMoveType(type);
+						moveArray[i].setMoveCategory(category);
+						moveArray[i].setMoveDmg(myDmg);
+						moveExists = true; 
+					}
+				}
+
+				// if the new MOVE does NOT already exist in the collection, then add the new MOVE to the array. 
+				if(moveExists == false) 
+				{
+					Move newMove = new Move(name, type, category, myDmg);
+					moveArray[numMoves++] = newMove;
```

Comment:
i think you still need to do a numMoves++; line of code after here since moveArray[numMoves++] will only evaluate that statement for the array and not your actual global variable.

Reply:
Line 72 is reading in a new move that doesn't exist yet in the move collection.     

// movedata.txt 
Shadow Ball, ... , ... , and so on
Tackle, ... , ... , and so on 

numMoves = 0; 
the moveArray is fresh and has no moves yet. 

So when this section of code happens, 
it will be moveArray[0] = Shadow Ball...
moveArray[1] = Tackle ...